### ISSUE TITLE
torch.linalg.cond return dtype inconsistent with doc and np.linalg.cond

### ISSUE THREAD

**Issue by heitorschueroff** [pytorch/pytorch#51643]
## 🐛 Bug

torch.linalg.cond always returns a real-valued tensor which is inconsistent with the doc and np.linalg.cond.

## To Reproduce

From the doc examples:

```
import torch
a = torch.randn(3, 4, 4, dtype=torch.complex64)
torch.linalg.cond(a, 1)
```

Result has dtype ``float32`` instead of ``complex64``.

## Expected behavior

Result dtype should match input dtype except when p = {None, 2, -2} and out=None, in which case the result should be real valued.

## Environment

PyTorch version: 1.8.0a0+5cf3278
Is debug build: False
CUDA used to build PyTorch: 10.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.1 LTS (x86_64)
GCC version: (GCC) 8.4.0
Clang version: Could not collect
CMake version: version 3.14.0

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 10.1.105
GPU models and configuration: 
GPU 0: Quadro GP100
GPU 1: Quadro GP100

Nvidia driver version: 450.80.02
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

cc @ezyang @anjali411 @dylanbespalko @mruberry @jlin27 @jianyuh @nikitaved @pearu @heitorschueroff @walterddr @IvanYashchuk

**Comment by heitorschueroff** [comment]
@mruberry and @IvanYashchuk, do we want to respect the out dtype for p = {None, 2, -2} or should we require out to be real valued for these?

**Comment by IvanYashchuk** [comment]
`torch.linalg.cond` was changed to always return a real-valued tensor in https://github.com/pytorch/pytorch/pull/48284. We've discussed that with @kurtamohler and @mruberry yesterday and came to the conclusion it's the right thing to be divergent from NumPy in this case. This behavior should be fixed in NumPy and the issue was filed https://github.com/numpy/numpy/issues/18304.

When normal `torch.linalg.cond` is called it should return the real number always (float32 for complex64 inputs, float64 for complex128 inputs). When the out variant is called float and complex tensors should be allowed according to the description of the correct "out=" behavior https://github.com/pytorch/pytorch/wiki/Developer-FAQ#how-does-out-work-in-pytorch.
